label_studio/data_import/api.py

```diff
@@ -712,3 +696,51 @@ def get(self, request, *args, **kwargs):
         response.headers['Cache-Control'] = f'no-store, max-age={max_age}'
 
         return response
+
+
+class TaskPresignStorageData(PresignAPIMixin, APIView):
+    """A file proxy to presign storage urls at the task level."""
+
+    swagger_schema = None
+    http_method_names = ['get']
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """Get the presigned url for a given fileuri"""
+        request = self.request
+        task_id = kwargs.get('task_id')
+        fileuri = request.GET.get('fileuri')
+
+        if fileuri is None or task_id is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Not sure 404 is the right status code for a request with missing data.  A more appropriate response code would be 400 bad request (generally due to client error - in this case missing params)